### ISSUE TITLE
[1.4] Host & Play doesn't work on unix.

### ISSUE THREAD

**Issue by Mirsario** [tModLoader/tModLoader#1595]
May be due to steam networking being selected
```System.ComponentModel.Win32Exception (2): No such file or directory
   at System.Diagnostics.Process.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at Terraria.ModLoader.Core.LoggingHooks.<>c.<HookProcessStart>b__1_0(Func`2 orig, Process self) in tModLoader\Terraria\ModLoader\Core\LoggingHooks.cs:line 21
   at Terraria.Social.Steam.NetClientSocialModule.LaunchLocalServer(Process process, ServerMode mode) in tModLoader\Terraria\Social\Steam\NetClientSocialModule.cs:line 72
   at Terraria.Main.DrawMenu(GameTime gameTime) in tModLoader\Terraria\Main.cs:line 36971
   at Terraria.Main.DoDraw(GameTime gameTime) in tModLoader\Terraria\Main.cs:line 48055
   at Terraria.Main.Draw(GameTime gameTime) in tModLoader\Terraria\Main.cs:line 47660
   at Microsoft.Xna.Framework.Game.Tick()
   at Microsoft.Xna.Framework.Game.RunLoop()
   at Microsoft.Xna.Framework.Game.Run()```

**Comment by Solxanich** [comment]
Supplying a log of someone who can't host&play on Linux to add to this.
Notable line I notice of issue is "System.Diagnostics.Process.GetModules()" as it appears in all errors in the log as the start of breaking.
[client (1).log](https://github.com/tModLoader/tModLoader/files/7157224/client.1.log)

**Comment by Solxanich** [comment]
Fixed in #2084
